documentation/create_component/add_a_metric.qmd

```diff
@@ -116,36 +117,50 @@ Please make sure that the following fields in the `functionality` and `functiona
 
 <!-- TODO: think about how to make this listing easier on the eye -->
 
-#### `functionality.name`
-
-A unique identifier for the metric component. Must be written in snake case. Example: `my_new_metric`. This will be the same as the name given in the `--name` argument of the  `create_component` command above.
-
-
-#### `functionality.info.metrics[]`
-
-* `.name`: A unique identifier for the metric (if only 1 metric in the component can be the same as `functionality.name`). Must be written in snake case. Example: `my_new_metric`.
-
-* `.pretty_name`: A label for the metric used for visualisations and documentation. Example: `"My new metric"`.
-
-* `.summary`: A one sentence summary of the metric. Used for creating short overviews of the components in a task.
-
-* `.description`: An explanation for how the metric works. Used for creating reference documentation of a task.
-
-* `.reference`: A bibtex reference key to the paper where the metrics is used.
-
-* `.documentation_url`: The url to the documentation of the metrics used.
-
-* `.repository_url`: The repository url for the metrics used.
-
-* `.min`: The minimum value of the metrics (`-inf` if there isn't a min).
-
-* `.max`: The maximum value of the metrics (`+inf` if there isn't a max).
-
-* `.maximize`: Set to `true` if a higher value is better and `false` if a lower value is better.
-
-#### `__merge__`
-
-The file specified in this field contains information regarding the input and output arguments of the component, as well as a unit test to ensure that the component is functioning properly. Normally you don't need to change this if you gave the right arguments to the `create_component` component.
+```{r echo=FALSE, warning=FALSE, error=FALSE}
+info <- read_and_merge_yaml("src/common/schemas/task_metric.yaml")
+
+process <- function(info, prefix = "", required = NA) {
+  type <- 
+    if ("type" %in% names(info)) {
+      info$type
+    } else if ("enum" %in% names(info)) {
+      "enum"
+    } else {
+      "unknown"
+    }
+  if (type == "object") {
+    prop_names <-
+      if (is.na(required)) {
+        names(info$properties)
+      } else if (required) {
+        info$required
+      } else {
+        setdiff(names(info$properties), info$required)
+      }
+    unlist(lapply(
+      prop_names,
+      function(prop_name) {
+        prop <- info$properties[[prop_name]]
+        process(prop, prefix = paste0(prefix, ".", prop_name), required = required)
+      }
+    ))
+  } else if (type == "array") {
+    c() # ignore for now
+  } else if (is.list(info)) {
+    paste0(
+      "#### `", prefix, "`\n",
+      "\n",
+      info$description
+    )
+  } else {
+    c()
+  }
+}
+info_str <- paste(process(info, required = TRUE), collapse = "\n\n")
+
+knitr::asis_output(info_str)
+```
```

Comment:
Update on WIP code: the required metadata fields is now empty: https://update-task-docs--openproblems.netlify.app/documentation/create_component/add_a_metric.html

Reply:
fixed but need to update schemas to make it work properly